crates/core/src/kernel/snapshot/mod.rs

```diff
@@ -956,4 +986,60 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_field_with_name() {
+        let schema = StructType::new(vec![
+            StructField::new("a", DataType::STRING, true),
+            StructField::new("b", DataType::INTEGER, true),
+        ]);
+        let field = get_stats_field(&schema, "b").unwrap();
+        assert_eq!(*field, StructField::new("b", DataType::INTEGER, true));
+    }
+
+    #[test]
+    fn test_field_with_name_nested() {
+        let nested = StructType::new(vec![StructField::new(
+            "nested_struct",
+            DataType::BOOLEAN,
+            true,
+        )]);
+        let schema = StructType::new(vec![
+            StructField::new("a", DataType::STRING, true),
+            StructField::new("b", DataType::Struct(Box::new(nested)), true),
+        ]);
+
+        let field = get_stats_field(&schema, "b.nested_struct").unwrap();
+
+        assert_eq!(
+            *field,
+            StructField::new("nested_struct", DataType::BOOLEAN, true)
+        );
+    }
+
+    #[test]
+    fn test_field_with_last_name_nested_backticks() {
```

Comment:
I'm not sure if we need to support nested enclosed field names vs the whole string being enclosed, but I would defer to someone more familiar with the protocol.

Reply:
Can you please mention that person? I don't know who should it be.

Reply:
Yeah, I personally have never seen this in the wild, but it doesn't mean it's not there somewhere. The protocol itself doesn't give any hardline opinion about this type of thing, so I suppose it's fine to have it, but understand it's a edge case we'll probably not see often. If it greatly complicates the code I'd consider removing it, but otherwise it's prolly fine.

Reply:
I have seen the `s mentioned by @roeap on #2572 referencing the [jvm implementation](https://github.com/delta-io/delta/blob/4b102d34a2ce881b2a851b4c6cfbf2ab3ab5534f/spark/src/main/scala/org/apache/spark/sql/delta/DeltaConfig.scala#L549-L561).

Before I knew about that and before the kernel types were introduced in 0.18. I added support for nested fields in the dataskipping table config by using the `.` as the delimiter in #2519.

So we could just split on `.` (or use datafusion parser as suggested) like we were in 0.17 and leave #2572 open to consider the ` enclosure for special chars.